frontends/p4/removeReturns.h

```diff
@@ -41,6 +41,67 @@ class HasExits : public Inspector {
     void postorder(const IR::ReturnStatement *) override { hasReturns = true; }
 };
 
+/**
+This visitor rewrites code after an 'if' that does a jump to somewhere else (with
+a break/continue/exit/return statement in one branch) by moving the code after the
+if into the branch that does not jump.  A block that looks like
+
+    ..some code
+    if (pred) jump somewhere;
+    ..more code
+becomes:
+    ..some code
+    if (pred) {
+        jump somewhere;
+    } else {
+        ..more code
+
+it also deals with cases where the else is not empty (the 'more code' is appended), or
+the jump is in the else branch (adds code to the ifTrue branch).
+
+The purpose of this is to rewrite code such the DoRemoveReturns (and similar rewrites
+for other jump statements like exit or loop break/continue) will more likely not need to
+introduce a boolean flag and extra tests to remove those branches.
+
+precondition:
+    switchAddDefault pass has run to ensure switch statements cover all cases
+*/
+class MoveToElseAfterBranch : public Modifier {
+    /* the pass does not use (inherit from) ControlFlowVisitor, even though it is doing
+     * control flow analysis, as it turns out to be more efficient to do it directly here
+     * by overloading the branching constructs (if/switch/loops) and not cloning the visitor,
+     * because we *only* care about statements (not expressions) locally in a block (and
+     * not across calls), and we only have one bit of data (hasJumped flag). */
+
+    bool hasJumped = false;
+    /* Set to true after a jump, indicating the code point we're at is unreachable.
+     * Set back to false at label-like points where code might jump to */
```

Comment:
Why is the comment after the member? Might be best to put it before as a doxygen comment (`/// ...`).